### PR TITLE
misc chaos

### PR DESCRIPTION
**Pull request by jschueller**
reflect in the api that metamodels are built on data, not from functions, following yesterdays meeting:
- Deprecate FunctionalChaos(Function, ...) ctors, FunctionalChaosResult.getComposedModel, MetaModelResult.getModel
- Add sample ctors, accessors to MetaModelResult & MetaModelAlgorithm

### CODE REVIEW

**Review by sofianehaddad — APPROVED** [review]

File: python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_chaos_ishigami.py [hunk 1]
```diff
@@ -95,8 +95,7 @@ def plotXvsY(sampleX, sampleY, figsize=(15, 3)):
 multivariateBasis = ot.OrthogonalProductPolynomialFactory(
     [im.X1, im.X2, im.X3])
 selectionAlgorithm = ot.LeastSquaresMetaModelSelectionFactory()
-projectionStrategy = ot.LeastSquaresStrategy(
-    inputTrain, outputTrain, selectionAlgorithm)
+projectionStrategy = ot.LeastSquaresStrategy(selectionAlgorithm)
```
Comment (josephmure) [inline python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_chaos_ishigami.py:98]: Unless I am mistaken, you are using an documented constructor here (and at other places):
image

You seem to be using `LeastSquaresStrategy(approxAlgoImpFact)`.

Comment (jschueller) [reply python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_chaos_ishigami.py:98]: yes, added this one to the list

File: lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx [hunk 2]
```diff
@@ -43,14 +44,39 @@ MetaModelAlgorithm::MetaModelAlgorithm()
...
+  : MetaModelAlgorithm(inputSample, outputSample, Dirac(Point(inputSample.getDimension(), 0.0)))
+{
+  // Nothing to do
+}
+
+/* Constructor with parameters */
+MetaModelAlgorithm::MetaModelAlgorithm(const Sample & inputSample,
+                                       const Sample & outputSample,
+                                       const Distribution & distribution,
+                                       const Function & model)
```
Comment (regislebrun) [inline lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx:59]: I don't see the point of keeping a model if we have an output sample.

**Review by regislebrun — CHANGES_REQUESTED** [review]
If this PR is supposed to remove the model based parametrization of the meta-model algorithms, then I don't understand why we would have to specify both an output sample and a model in several places.

File: lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx (hunk 2 above)
Comment (jschueller) [reply lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx:59]: The model accessors are deprecated; we keep them for this release then remove them (and the attributes/ctors) in the new release.

**Comment by jschueller** [comment]
> If this PR is supposed to remove the model based parametrization of the meta-model algorithms, then I don't understand why we would have to specify both an output sample and a model in several places.

The model accessors are deprecated; we keep them for this release then remove them (and the attributes/ctors) in the new release.

**Comment by regislebrun** [comment]
Ok but why do we need to actually store the model? The setModel() accessor could simply emit a warning, and the getModel() accessor could build a DatabaseFunction on demand, no?

**Comment by sofianehaddad** [comment]
@jschueller is there a way to fix #1473 with this PR?

**Comment by jschueller** [comment]
Indeed it fixes #1473

**Review by josephmure — APPROVED** [review]

File: lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx [hunk 3]
```diff
@@ -43,14 +44,39 @@ MetaModelAlgorithm::MetaModelAlgorithm()
   // Nothing to do
 }
 
+/* Constructor with parameters */
+MetaModelAlgorithm::MetaModelAlgorithm(const Sample & inputSample,
+                                       const Sample & outputSample)
+  : MetaModelAlgorithm(inputSample, outputSample, Dirac(Point(inputSample.getDimension(), 0.0)))
```
Comment (regislebrun) [inline lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx:50]: Why did you choose a Dirac distribution here? A UserDefined one would have been more adapted from a statistical point of view. Is it related to the cost of constructing such a distribution?

**Review by regislebrun — APPROVED** [review]
Only a minor question, otherwise good to go for me

File: lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx (hunk 3 above)
Comment (jschueller) [reply lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx]: I dont think its actually used, but UserDefined is ok too

**Review by mbaudin47 — COMMENTED** [review]
I am sorry, but I did not find the time to perform the requested review.

**Pull request merged**